src/GovUk.Education.ExploreEducationStatistics.Common/Services/TableStorageService.cs

```diff
@@ -74,5 +103,30 @@ public async Task<IEnumerable<TElement>> ExecuteQueryAsync<TElement>(string tabl
 
             return results;
         }
+
+        private static async Task DeleteByPartitionKey(CloudTable table, string partitionKey)
```

Comment:
Could we make this a little more generic `DeleteAllBy` method, where you can pass in a `query` string instead? I might even suggest not appending any filter if you intend to delete everything (or have another method for that).

Reply:
It works like this currently because you can only batch delete rows of the same partitionKey.
I could get rid of batch deleting altogether which would allow a query and a query with no `.where(filter)`. I suppose we don't know a problem even exists yet with performance of deleting.

Reply:
Ah I see, that seems like such a weird limitation for these batch deletes. 

Fair enough anyway, wouldn't go to the trouble of undoing all the batch stuff given it's there already!